functions/Get-DBAUptime.ps1

```diff
@@ -0,0 +1,163 @@
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER WindowsCredential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER SQLOnly
+Excludes the windows server information
+
+.NOTES 
+Original Author: Stuart Moore (@napalmgram), stuart-moore.com
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>
+
+.LINK
+https://dbatools.io/Get-DBAUptime
+
+.EXAMPLE
+Get-DbaUptime -SqlServer SqlBox1\Instance2
+
+Returns an object with SQL Server start time, uptime as TimeSpan object, uptime as a string, and Windows host boot time, host uptime as TimeSpan objects and host uptime as a string for the sqlexpress instance on winserver
+
+.EXAMPLE
+Get-DbaUptime -SqlServer winserver\sqlexpress, sql2016
+
+Returns an object with SQL Server start time, uptime as TimeSpan object, uptime as a string, and Windows host boot time, host uptime as TimeSpan objects and host uptime as a string for the sqlexpress instance on host winserver  and the default instance on host sql2016
+	
+.EXAMPLE   
+Get-DbaUptime -SqlServer sqlserver2014a, sql2016 -SqlOnly
+
+Returns an object with SQL Server start time, uptime as TimeSpan object, uptime as a string for the sqlexpress instance on host winserver  and the default instance on host sql2016
+
+.EXAMPLE   
+Get-SqlRegisteredServerName -SqlServer sql2014 | Get-DbaUptime 
+
+Returns an object with SQL Server start time, uptime as TimeSpan object, uptime as a string, and Windows host boot time, host uptime as TimeSpan objects and host uptime as a string for every server listed in the Central Management Server on sql2014
+	
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[Alias("SqlCredential")]
+		[PsCredential]$Credential,
+		[PsCredential]$WindowsCredential,
+		[Switch]$SQLOnly
+
+	)
+
+    	BEGIN
+	{
+        $functionname = "Get-DBAUptime"
+		$collection = New-Object System.Collections.ArrayList
+	}
+    	PROCESS
+	{
+		$servercount = ++$i
+		foreach ($servername in $SqlServer)
+		{
+            write-verbose "$functionname - server = $servername"
+			try
+			{
+				$server = Connect-SqlServer -SqlServer "$servername" -SqlCredential $Credential
+			}
+			catch
+			{
+				if ($servercount -eq 1)
```

Comment:
This is the old way to catch a failed connection.
We changed to:
```
		Write-Verbose "Connecting to $SqlServer"
		try
		{
			$server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $Credential -ErrorVariable ConnectError
			
		}
		catch
		{
			Write-Warning $_
			continue
		}
```
in newer functions